main/src/library/DnsWithResult.flix

```diff
@@ -0,0 +1,96 @@
+/*
+ *  Copyright 2025 Cade Lueker
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+///
+/// an effect for resolving a domain name to an ip address
+///
+eff DnsWithResult {
+    def lookup(host: String): Result[IoError, IpAddr]
+    def lookupAll(host: String): Result[IoError, List[IpAddr]]
+}
+
+mod DnsWithResult {
+    use IoError.ErrorKind
+    use IoError.IoError
+
+    import java.io.IOException
+    import java.lang.IllegalArgumentException
+    import java.net.InetAddress
+    import java.net.UnknownHostException
+
+    ///
+    /// Converts a `List[Int8]` into `IpAddr`
+    ///
+    /// an Ipv4 is 32 bytes and Ipv6 is 128 bytes
+    ///
+    def bytesToIpAddr(l: List[Int8]): IpAddr = match l {
```

Comment:
Is this functionality duplicated in Dns and DnsWithResult? Perhaps we move it into IpAddr then?